### PR TITLE
fix(jsonrpsee): less deps when defining RPC API.

### PR DESCRIPTION
**Pull request by niklasad1**
Closing https://github.com/paritytech/jsonrpsee/issues/847 by introducing two new features `server-core` and `client-core` to only import relevant types for defining the RPC API i.e, not import transport related deps.

It makes things better such that if one doesn't want the transport stuff just the `client` or `server` types from core it reduces the dependency tree a lot. The typical use-case if some third-party crate or library just defines the RPC API and some other crate such a client application just want to re-use the API definition.

### CODE REVIEW

File: jsonrpsee/Cargo.toml [hunk 1]
```diff
@@ -34,9 +35,11 @@ ws-client = ["jsonrpsee-ws-client", "jsonrpsee-types", "jsonrpsee-core"]
 ws-server = ["jsonrpsee-ws-server", "jsonrpsee-types", "jsonrpsee-core"]
 macros = ["jsonrpsee-proc-macros", "jsonrpsee-types", "jsonrpsee-core/client", "tracing"]
 
-client = ["http-client", "ws-client", "wasm-client"]
-server = ["http-server", "ws-server"]
-full = ["client", "server", "macros", "async-client", "client-ws-transport"]
+client-full = ["client", "http-client", "ws-client", "wasm-client", "client-ws-transport", "client-web-transport", "async-client"]
+client = ["macros"]
```
Comment (niklasad1) [inline jsonrpsee/Cargo.toml:39]: I guess we could rename this to client-core as well and keep the client feature, not sure what's the best

**Review by lexnv — APPROVED** [review]
LGTM!

File: jsonrpsee/Cargo.toml (hunk 1 above)
Comment (jsdw) [reply jsonrpsee/Cargo.toml:39]: I think I prefer "client" being kept as-is and "client-core" for those "optimisation" cases where you want less. Also it wouldn't be a breaking change I guess that way so it's a win win really :)

Comment (niklasad1) [reply jsonrpsee/Cargo.toml:39]: fair enough, I'm convinced

**Review by TarikGul — APPROVED** [review]

**Pull request merged**